lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -367,7 +368,12 @@ def add_stored_credential(xml, options)
 
       def merchant_initiated_unscheduled(options)
         return options[:merchant_initiated_unscheduled] if options[:merchant_initiated_unscheduled]
-        return 'Y' if options.dig(:stored_credential, :initiator) == 'merchant' && options.dig(:stored_credential, :reason_type) == 'unscheduled'
+        return 'Y' if options.dig(:stored_credential, :initiator) == 'merchant' && options.dig(:stored_credential, :reason_type) == 'unscheduled' || options.dig(:stored_credential, :reason_type) == 'recurring'
+      end
+
+      def recurring_transaction(options)
+        return options[:ssl_add_token] if options.dig(:stored_credential, :reason_type) == 'recurring' && options.dig(:store_credential, :initial_transaction) == true
```

Comment:
I think this will produce an error if `options[:ssl_add_token]` isn't set.

Reply:
Never mind, I see now that you're returning `false` if that condition is met. So, no error.

Reply:
I'm not seeing any failures from calls without it being set from Elavon currently. Other tests are sending without this parameter being set and it isn't causing an error.

Reply:
Ok, maybe this won't lead to an error, but if `options[:ssl_add_token]` isn't set, it will just return nil/empty. The gateway seems to handle it, but it might be the case if the user doesn't pass this value, the transaction might not behave how they want/expect.

Reply:
The field will not send it unless they are sending it in as a recurring, anything else will not include the field at all. With my latest commit it will either send a 'Y' or 'N' depending on if it is an initial transaction or not, both are accepted values for the gateway. In my prior code, it would only send add_ssl_token = 'Y' no matter what kind of recurring transaction it was, it did not send for non- recurring at all.